### PR TITLE
Use `assert_writable_eq` in units formatting test cases

### PR DESCRIPTION
**Pull request by younies**

### CODE REVIEW

File: components/experimental/src/dimension/units/format.rs [hunk 1]
```diff
@@ -53,35 +53,47 @@ impl<'l> Writeable for FormattedUnit<'l> {
     }
 }
 
+impl core::fmt::Display for FormattedUnit<'_> {
```
Comment (younies) [inline components/experimental/src/dimension/units/format.rs:56]: This is required by `assert_writeable_eq!` macro, or  I need to implement `core::fmt::ToString` trait

Comment (robertbastian) [reply components/experimental/src/dimension/units/format.rs:56]: use https://docs.rs/writeable/latest/writeable/macro.impl_display_with_writeable.html

File: components/experimental/src/dimension/units/format.rs [hunk 2]
```diff
@@ -53,35 +53,47 @@ impl<'l> Writeable for FormattedUnit<'l> {
...
+    let (locale, meter) = (locale!("en-US").into(), "meter");
     let fmt = UnitsFormatter::try_new(&locale, meter, Default::default()).unwrap();
     let value = "12345.67".parse().unwrap();
-    let formatted_unit = fmt.format_fixed_decimal(&value, meter);
-    let mut sink = String::new();
-    formatted_unit.write_to(&mut sink).unwrap();
-    assert_eq!(sink.as_str(), "12,345.67 m");
+    assert_writeable_eq!(fmt.format_fixed_decimal(&value, meter), "12,345.67 m");
+
+    let (locale, century) = (locale!("en-US").into(), "century");
```
Comment (robertbastian) [inline components/experimental/src/dimension/units/format.rs:76]: I don't understand why you're tupling these up. If anything, the unit should be combined with the value:

```rust
let (value, unit) = ("12345.67".parse().unwrap(), "century");
```

File: components/experimental/src/dimension/units/format.rs [hunk 3]
```diff
@@ -53,35 +53,47 @@ impl<'l> Writeable for FormattedUnit<'l> {
...
+    let value = "12345.67".parse().unwrap();
+    let formatted_unit = fmt.format_fixed_decimal(&value, century);
+    assert_writeable_eq!(formatted_unit, "12,345.67 centuries");
 
-    let locale = locale!("de-DE").into();
-    let meter = "meter";
+    let (locale, meter) = (locale!("de-DE").into(), "meter");
     let fmt = UnitsFormatter::try_new(&locale, meter, Default::default()).unwrap();
     let value = "12345.67".parse().unwrap();
     let formatted_unit = fmt.format_fixed_decimal(&value, meter);
```
Comment (robertbastian) [inline components/experimental/src/dimension/units/format.rs:93]: nit: it's not a formatted unit, it's a formatted value that has a unit

Comment (younies) [reply components/experimental/src/dimension/units/format.rs]: done

File: components/experimental/src/dimension/units/format.rs (hunk 2 above)
Comment (younies) [reply components/experimental/src/dimension/units/format.rs]: done

File: components/experimental/src/dimension/units/format.rs [hunk 4]
```diff
@@ -53,47 +53,52 @@ impl<'l> Writeable for FormattedUnit<'l> {
...
-    let meter = "meter";
-    let fmt = UnitsFormatter::try_new(&locale, meter, Default::default()).unwrap();
-    let value = "12345.67".parse().unwrap();
-    let formatted_unit = fmt.format_fixed_decimal(&value, meter);
-    let mut sink = String::new();
-    formatted_unit.write_to(&mut sink).unwrap();
-    assert_eq!(sink.as_str(), "12.345,67 m");
+    let test_cases = [
+        (
+            locale!("en-US").into(),
```
Comment (robertbastian) [inline components/experimental/src/dimension/units/format.rs:68]: nit: move the `into`s into the loop

**Review by robertbastian — APPROVED** [review]

File: components/experimental/src/dimension/units/format.rs (hunk 4 above)
Comment (younies) [reply components/experimental/src/dimension/units/format.rs:68]: Done here: https://github.com/unicode-org/icu4x/pull/5234

**Pull request merged**